### PR TITLE
[Impeller Scene] Add offline mesh importer

### PR DESCRIPTION
**Pull request by bdero**
* scene_importer lib and executable. Everything is in place aside from actually converting the GLTF data. Structured with the assumption that we may add importers for more formats in the future.
* GN jazz to run the importer.
* Simple flatbuffer with only a single mesh, no bone transforms/weights or animations right now -- it will be appropriate to extend this into a more comprehensive scene description down the road.
* Example GLTF binary mesh fixture containing a 3D Flutter logo with bezeled edges. This is a simplified version of the PBR textured mesh I showed off a couple of times in the past. Exported with normals, tangents, colors, and a good quality UV unwrap. I'm not bothering to import vertex colors straight away, but it'll come in handy for testing stuff later on.

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 4.
View them at https://flutter-engine-gold.skia.org/cl/github/37981

File: impeller/scene/importer/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    # target.
+    # TODO(bdero): Fix tinygltf compilation warnings.
+    #"//third_party/tinygltf",
+  ]
+}
+
+generated_file("scene_importer_license") {
+  source_path = rebase_path(".", "//flutter")
+  git_url = "https://github.com/flutter/engine/tree/$engine_version"
+  outputs = [ "$target_gen_dir/LICENSE.impellerc.md" ]
```
Comment (zanderso) [inline impeller/scene/importer/BUILD.gn:50]:
```suggestion
  outputs = [ "$target_gen_dir/LICENSE.scene_importer.md" ]
```

File: impeller/scene/importer/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    "## Licenses",
+    "",
+    "### scene_importer",
+    "",
+    read_file("//flutter/sky/packages/sky_engine/LICENSE", "string"),
+    "",
+
+    # These licenses are ignored by the main license checker, since they are not
+    # shipped to end-application binaries and only shipped as part of developer
+    # tooling in impellerc. Add them here.
```
Comment (zanderso) [inline impeller/scene/importer/BUILD.gn:67]:
```suggestion
    # tooling in scene_importer. Add them here.
```

File: impeller/scene/importer/BUILD.gn [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  ]
+
+  public_deps = [
+    ":importer_flatbuffers",
+    "../../base",
+    "../../compiler:utilities",
+    "../../geometry",
+    "//flutter/fml",
+
+    # All third_party deps must be reflected below in the impellerc_license
```
Comment (zanderso) [inline impeller/scene/importer/BUILD.gn:40]:
```suggestion
    # All third_party deps must be reflected below in the scene_importer_license
```

File: impeller/tools/impeller.gni [hunk 4]
```diff
@@ -613,3 +618,47 @@ template("impeller_shaders") {
     }
   }
 }
+
+# Dispatches to the build or prebuilt impellerc depending on the value of
```
Comment (zanderso) [inline impeller/tools/impeller.gni:622]:
```suggestion
# Dispatches to the build or prebuilt scene_importer depending on the value of
```

File: impeller/scene/importer/BUILD.gn [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    # tooling in impellerc. Add them here.
+    "## Additional open source licenses",
+    "",
+    "### tinygltf",
+    "",
+    read_file("//third_party/tinygltf/LICENSE", "string"),
+  ]
+}
+
+group("importer") {
```
Comment (chinmaygarde) [inline impeller/scene/importer/BUILD.gn:76]: Nit: On naming, perhaps just `scenec` like `impellerc`. `importer` is rather generic.

File: impeller/scene/importer/mesh.fbs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+struct Vertex {
+  position: Vec3;
+  normal: Vec3;
+  tangent: Vec3;
+  texture_coords: Vec2;
+}
+
+table Mesh {
```
Comment (chinmaygarde) [inline impeller/scene/importer/mesh.fbs:31]: We'll probably have to sketch this out some more but this is a start!

File: impeller/scene/importer/BUILD.gn (hunk 5 above)
Comment (bdero) [reply impeller/scene/importer/BUILD.gn:76]: Ah yeah, `scenec` seems good. My mind kept getting stuck on `geometryc` which I didn't like very much.

File: impeller/scene/importer/mesh.fbs (hunk 6 above)
Comment (bdero) [reply impeller/scene/importer/mesh.fbs:31]: Indeed, one triangle mesh to fake it till we make it. Materials, fuller scene description, bones, weights, and animation will all be necessary soon in my estimation. :)

**Comment by chinmaygarde** [comment]
May I land this?

**Comment by bdero** [comment]
Yeah, I'm in the middle of renaming everything to be SceneC instead of "Scene Importer" at the moment, but I can just land that as a separate patch later today.

**Pull request merged**